lib/Interpreter/CMakeLists.txt

```diff
@@ -17,6 +17,13 @@ if(EMSCRIPTEN)
     LINK_LIBS
     clangInterpreter
   )
+  #FIXME: This is needed until https://github.com/emscripten-core/emscripten/blob/ac676d5e437525d15df5fd46bc2c208ec6d376a3/cmake/Modules/Platform/Emscripten.cmake#L36
+  # is patched out of emsdk, as --soname is not recognised by emscripten.
+  #FIXME: A patch is needed to llvm to remove -Wl,-z,defs since it is now recognised on emscripten. What needs to be removed is here
+  # https://github.com/llvm/llvm-project/blob/128e2e446e90c3b1827cfc7d4d19e3c0976beff3/llvm/cmake/modules/HandleLLVMOptions.cmake#L318
```

Comment:
I was thinking after we open PRs to the main repos but that's also probably fine...